### PR TITLE
LPS-134121 allow passing bodies in delete and string arrays

### PR DESCRIPTION
**Pull request by nhpatt**
Remove restrictions that do not allow passing bodies in DELETE and GET requests and change client generation logic for adding body content in the request to ignore path parameters.

/@pei-jung

### CODE REVIEW

**Comment by nhpatt** [comment]
ci:test:sf

**Comment by nhpatt** [comment]
ci:test:relevant

File: modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/client_resource.ftl [hunk 1]
```diff
@@ -199,7 +199,7 @@ public interface ${schemaName}Resource {
 			public HttpInvoker.HttpResponse ${javaMethodSignature.methodName}HttpResponse(${parameters}) throws Exception {
 				HttpInvoker httpInvoker = HttpInvoker.newHttpInvoker();
 
-				<#if freeMarkerTool.hasHTTPMethod(javaMethodSignature, "patch", "post", "put")>
+				<#if freeMarkerTool.hasHTTPMethod(javaMethodSignature, "delete", "patch", "post", "put")>
```
Comment (javierdearcos) [inline modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/client_resource.ftl:202]: Should we allow the "get" also here? (I think that in that case should be removing the if) As we said that the body in a GET method is also allowed

**Review by javierdearcos — APPROVED** [review]

File: modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/client_resource.ftl (hunk 1 above)
Comment (nhpatt) [reply modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/client_resource.ftl:202]: Yeah, but we are not using it internally and it has more drawbacks that the rest of the methods (it creates URLs that are not bookmarkable) so I prefer to track usage this way.

**Comment by nhpatt** [comment]
ci:forward

**Pull request closed without merging**